Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -949,6 +1001,15 @@ public struct ObjectId: BSONValue, Equatable, CustomStringConvertible, Codable {
     }
 }
 
+// An extension of `ObjectId` to add the capability to be hashed
+extension ObjectId: Hashable {
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(hex)
```

Comment:
libbson provides a [way](http://mongoc.org/libbson/current/bson_oid_hash.html) to generate hashes for `bson_oid_t`'s already, so I think it makes sense to leverage that.

If you look at the `timestamp` computed property, you'll see an example of how we use the underlying libmongoc type. 

In this case, we could do something like: 
```swift
let hashValue = withUnsafePointer(to: self.oid) { oid in
    return bson_oid_hash(oid)
}
hasher.combine(hashValue)
```